packages/protocol/contracts/bridge/libs/LibBridgeRelease.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.9;
+
+import "./LibBridgeData.sol";
+import "./LibBridgeStatus.sol";
+
+/**
+ * @author dantaik <dan@taiko.xyz>
+ */
+library LibBridgeRelease {
+    // using LibAddress for address;
+    using LibBridgeData for IBridge.Message;
+
+    event EtherReleased(bytes32 indexed msgHash, address to, uint256 amount);
+
+    function releaseEther(
+        LibBridgeData.State storage state,
+        AddressResolver resolver,
+        IBridge.Message calldata message,
+        bytes calldata proof
+    ) internal {
+        require(message.owner != address(0), "B:owner");
+        require(message.srcChainId == block.chainid, "B:srcChainId");
+
+        bytes32 msgHash = message.hashMessage();
+        require(state.etherReleased[msgHash] == false, "B:etherReleased");
+        require(
+            LibBridgeStatus.isMessageFailed(
+                resolver,
+                msgHash,
+                message.destChainId,
+                proof
+            ),
+            "B:notFailed"
+        );
+
+        state.etherReleased[msgHash] = true;
+
+        uint256 releaseAmount = message.depositValue + message.callValue;
+
+        if (releaseAmount > 0) {
+            (bool success, ) = message.owner.call{value: releaseAmount}("");
```

Comment:
If this is the L2 bridge, shall we resolve the `EtherVault` address here, then let it send `releaseAmount` ETHs? instead of sending it directly from the L2 bridge.

Reply:
Fixed